### PR TITLE
Feature/suggest ens domain

### LINKED ISSUE #954: Suggest ENS domain for logged users
On the setup page here https://snapshot.org/#/setup
We should check if the logged user has ENS domains, if he has we should show max 5 suggestions like this:
image
When we click on the suggested domain it should populate the field. The existing spaces should not appear as suggested.

### PR DESCRIPTION
**Pull request by sarmatdev**
Fixes #954.

Changes proposed in this pull request:
- Setup graphQL for ENS
- Show user ENS names as suggestions
-

### CODE REVIEW

**Comment by sarmatdev** [comment]
There is my draft PR

How do I need to handle the loading status of fetching names?
What message do I need to show the user if he doesn't have any ENS names, and how it should look?

**Comment by mktcode** [comment]
I would require wallet connection at the beginning. Then you can look for ENS domains and display them to chose from, plus a "Register Domain" button that just links to the ENS website and later we implement registration through snapshot interface. Text input is not needed at all I think.

**Comment by sarmatdev** [comment]
@bonustrack What would you say?

**Comment by sarmatdev** [comment]
Added some improvements

**Comment by ChaituVR** [comment]
@sarmatdev resolve conflicts plz

**Review by ChaituVR — CHANGES_REQUESTED** [review]
Doesn't seem to be working, always shows me:
```
It looks like your address does not own any ENS name
```

and also revert changes from yarn.lock

**Comment by mktcode** [comment]
image

File: src/composables/useEns.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+export function useEns() {
+  const { web3 } = useWeb3();
+  const { ensApolloQuery } = useApolloQuery();
+
+  async function getEnsNames() {
+    const res = await ensApolloQuery({
+      query: ENS_QUERY,
+      variables: {
+        id: web3.value.account
```
Comment (mktcode) [inline src/composables/useEns.ts:13]:
```suggestion
        id: web3.value.account.toLowerCase()
```

File: src/views/Setup.vue [hunk 2]
```diff
@@ -1,10 +1,29 @@
 <script setup>
 import { ref } from 'vue';
 import { useRouter } from 'vue-router';
+import { useEns } from '@/composables/useEns';
```
Comment (mktcode) [inline src/views/Setup.vue:4]:
```suggestion
import { ref, computed, watch } from 'vue';
import { useRouter } from 'vue-router';
import { useEns } from '@/composables/useEns';
import { useWeb3 } from '@/composables/useWeb3';

const { web3 } = useWeb3();
const web3Account = computed(() => web3.value.account);
```

**Review by mktcode — CHANGES_REQUESTED** [review]
With the changes I suggest it works for me.

I used this pattern, which I see in a lot of components.
```
const { web3 } = useWeb3();
const web3Account = computed(() => web3.value.account);
watch(web3Account, ...);
```

But `web3Account` can actually be moved to `useWeb3`, so that this becomes:
```
const { web3Account } = useWeb3();
watch(web3Account, ...);
```

Can be changed in a successive pull request.

File: src/views/Setup.vue [hunk 3]
```diff
@@ -1,10 +1,29 @@
...
+function loadEns() {
+  loading.value = true;
+  getEnsNames()
+    .then(res => {
+      domains.value = res.account?.domains.slice(0, 5);
+    })
+    .finally(() => (loading.value = false));
+}
+
+loadEns();
```
[the remark this answers is not in the document]
Comment (mktcode) [reply src/views/Setup.vue:26]: Why is this marked as resolved?

**Review by mktcode — APPROVED** [review]

File: src/views/Setup.vue [hunk 4]
```diff
@@ -1,10 +1,33 @@
...
+function loadEns() {
+  loading.value = true;
+  getEnsNames()
+    .then(res => {
+      domains.value = res.account?.domains.slice(0, 5);
+    })
+    .finally(() => (loading.value = false));
+}
+
+watch(web3Account, loadEns);
```
Comment (mktcode) [inline src/views/Setup.vue:30]:
```suggestion
loadEns();
watch(web3Account, loadEns);
```

To make sure ENS names are loaded when entering the setup page and not ONLY when web3Account changes.

**Comment by mktcode** [comment]
Committed my changes. This is ready to be merged I'd say.

**Comment by bonustrack** [comment]
Cool, here is some suggestion
- Dont load suggestion if user is not logged
- Use a small circle loading and not this fat loading placeholder
image
- Don't show "Looks like you dont have ENS domains", now i can see it even if request to query ENS domain is not resolved yet, I think we can just remove it

Maybe a better way to imagine this would be like this:

- If user is not logged, just show a "Connect wallet" button like this:
image

- If user is logged but don't have any domain, show a page to register a space like this:
image
The next button would redirect to ENS website

- If user is logged and have domain show the domains as buttons and before the form like this:

image

**Comment by mktcode** [comment]
@sarmatdev I'll take over from here, if that's ok for you. :)

**Comment by sarmatdev** [comment]
> @sarmatdev I'll take over from here, if that's ok for you. :)

Yes, do it if you are want

**Comment by mktcode** [comment]
Closing in favor of: ...

**Pull request closed without merging**